packages/flutter_driver/lib/src/driver/driver.dart

```diff
@@ -706,6 +656,18 @@ class FlutterDriver {
     //       The two-second gap should be long enough for the GPU thread to
     //       finish rasterizing the frame, but not longer than necessary to keep
     //       driver tests as fast a possible.
+    //
+    //       -------------------------------------------------------------------
+    //       With this timeout, if we're not lucky:
+    //       -------------------------------------------------------------------
+    //       UI    : <-- build -->
+    //       GPU   :               <-- rasterize randomly slow today -->
+    //       Gap   :              |    2 seconds or more   |
+    //       Driver:                                        <-- screenshot -->
+    //
+    //       In practice, sometimes the device gets really busy for a while and
+    //       even two seconds isn't enough, which means that this is still racy
+    //       and a source of flakes.
     await Future<void>.delayed(const Duration(seconds: 2));
 
     final Map<String, dynamic> result = await _peer.sendRequest('_flutter.screenshot').timeout(timeout);
```

Comment:
We're not guaranteed that `timeout` is set here

Reply:
Just remove the `.timeout()` call.

Reply:
Removed (also removed the argument to the method).